### PR TITLE
[rllib] Changes for preprocessors

### PR DESCRIPTION
**Pull request by richardliaw**
Addresses #1027, introduces preprocessing in configuration. 

This change causes A3C to converge at the following rate:
```
Gradient steps per Iteration: 1000
Env: PongDeterminsitic-v4
Mean episode score: >19.0
16 workers
~850 seconds
```

cc @pcmoritz @ericl

### CODE REVIEW

File: python/ray/rllib/a3c/a3c.py [hunk 1]
```diff
@@ -12,13 +12,18 @@
...
 
 
 DEFAULT_CONFIG = {
     "num_workers": 4,
     "num_batches_per_iteration": 100,
-    "batch_size": 10
+    "batch_size": 10,
+    "model": { "grayscale": True,
+               "zero_mean": False,
+               "dim":42,
```
Comment (ericl) [inline python/ray/rllib/a3c/a3c.py]: nit: space before 42 (does this pass lint?)

File: python/ray/rllib/models/catalog.py [hunk 2]
```diff
@@ -14,11 +14,15 @@
 
 MODEL_CONFIGS = [
     "conv_filters",
-    "downscale_factor",
+    # TODO(rliaw) introduce specific ATARI config
```
Comment (ericl) [inline python/ray/rllib/models/catalog.py]: I think you might need a colon after the TODO(rliaw)

File: python/ray/rllib/models/catalog.py [hunk 3]
```diff
@@ -14,11 +14,15 @@
...
+    "dim",
+    "grayscale",
+    "zero_mean",
     "extra_frameskip",
     "fcnet_activation",
     "fcnet_hiddens",
-    "free_log_std"
+    "free_log_std",
+    "use_lstm"
```
Comment (ericl) [inline python/ray/rllib/models/catalog.py]: Move this to the A3C conf, this is never used in catalog.

File: python/ray/rllib/models/preprocessors.py [hunk 4]
```diff
@@ -24,20 +26,36 @@ def transform(self, observation):
 
 class AtariPixelPreprocessor(Preprocessor):
     def _init(self):
-        self.downscale_factor = self.options.get("downscale_factor", 2)
-        self.dim = int(160 / self.downscale_factor)
+        self.grayscale = self.options.get("grayscale", False)
```
Comment (ericl) [inline python/ray/rllib/models/preprocessors.py]: run_multinode_tests uses downscale factor currently, so you'll need to fix that for tests to pass

File: python/ray/rllib/models/catalog.py [hunk 5]
```diff
@@ -14,11 +14,15 @@
 
 MODEL_CONFIGS = [
     "conv_filters",
-    "downscale_factor",
+    # TODO(rliaw) introduce specific ATARI config
+    "dim",
+    "grayscale",
+    "zero_mean",
```
Comment (ericl) [inline python/ray/rllib/models/catalog.py]: Not a problem introduced by this PR, but could we add a short comment on what each of these config options do?

File: python/ray/rllib/models/preprocessors.py (hunk 4 above)
Comment (richardliaw) [reply python/ray/rllib/models/preprocessors.py]: Done.

**Pull request merged**